src/status_im2/contexts/chat/lightbox/text_sheet/view.cljs

```diff
@@ -33,50 +27,62 @@
             animations                (utils/init-animations overlay-opacity)
-            derived                   (utils/init-derived-animations animations)]
+            derived                   (utils/init-derived-animations animations)
+            expanding-message?        (> @text-height (* constants/line-height 2))]
         [gesture/gesture-detector
          {:gesture (utils/sheet-gesture animations
                                         expanded-height
                                         max-height
+                                        full-height
                                         overlay-z-index
                                         expanded?
-                                        dragging?)}
-         [reanimated/touchable-opacity
-          {:active-opacity 1
-           :on-press
-           #(utils/expand-sheet animations
-                                expanded-height
-                                max-height
-                                overlay-z-index
-                                expanded?
-                                text-sheet-lock?)
-           :style (style/sheet-container derived)}
-          [bar @text-height]
-          [reanimated/linear-gradient
-           {:colors [colors/neutral-100-opa-0 colors/neutral-100]
-            :start  {:x 0 :y 1}
-            :end    {:x 0 :y 0}
-            :style  (style/top-gradient animations insets)}]
-          [linear-gradient/linear-gradient
-           {:colors [colors/neutral-100-opa-50 colors/neutral-100-opa-0]
-            :start  {:x 0 :y 1}
-            :end    {:x 0 :y 0}
-            :style  style/bottom-gradient}]
-          [gesture/scroll-view
-           {:scroll-enabled        @expanded?
-            :scroll-event-throttle 16
-            :on-scroll             #(utils/on-scroll % @expanded? @dragging? animations)
-            :style                 {:height (- max-height constants/bar-container-height)}}
-           [message-view/render-parsed-text
-            {:content        content
-             :chat-id        chat-id
-             :style-override style/text-style
-             :on-layout      #(utils/on-layout % text-height)}]]]]))))
+                                        dragging?
+                                        expanding-message?)}
+         [gesture/gesture-detector
+          {:gesture (-> (gesture/gesture-tap)
+                        (gesture/enabled expanding-message?)
+                        (gesture/on-start (fn []
+                                            (utils/expand-sheet animations
+                                                                expanded-height
+                                                                max-height
+                                                                overlay-z-index
+                                                                expanded?
+                                                                text-sheet-lock?))))}
```

Comment:
NOTE: using the tap gesture instead of the `touchable-opacity`, since it was behaving weirdly i.e. not all taps were always recognised after the scroll changes. I guess it's due to the pointer events being hijacked by the pan handler.